fhevm/tee_bit.go

```diff
@@ -0,0 +1,145 @@
+package fhevm
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func teeShlRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
+		case 1:
```

Comment:
there's a case 0 too

Reply:
There is not Shl operation between ebool?

Reply:
Ah true. Could you add a comment here, while reading code I thought it was a bug, not intended

Reply:
Okay